HikVisionSource/README.md

```diff
@@ -0,0 +1,203 @@
+
+# Setting Up Your Vantiq Environment<a name="vantiq" id="vantiq"></a>
+
+An understanding of the VANTIQ Extension Source SDK is assumed. Please read the [Extension Source README.md](../README.md) for more information.
+
+In order to incorporate this Extension Source, you will need to create the Source Implementation in the Vantiq system.
+
+## Source Implementation
+
+When creating a HikVision Extension source, you must first create the source implementation. This is done by using the [*HikVisionImpl.json*](src/test/resources/HikVisionImpl.json) file found in src/test/resources/HikVisionImpl.json. To make the source type known to Vantiq, use the vantiq cli command
+
+```
+
+vantiq -s <profileName> load sourceimpls <fileName>
+```
+
+where <profileName> is replaced by the Vantiq profile name, and <fileName> is the file to be loaded.
+
+(You can, of course, change the source implementation name from that provided in this definition file.)
+
+That completed, you will need to create the Source in the Vantiq system.
+
+## Source Configuration
+
+To set up the Source in the Vantiq system, you will need to add a Source to your project.  Make sure you have properly added a Source Implementation to Vantiq. Once this is complete, you can select EasyModbus (or whatever you named your Source Definition) as the Source Type. You will then need to fill out the Source Configuration Document.
+
+The Configuration document may look similar to the following example:
+
+The user must [define the HikVision Source implementation](../README.md#-defining-a-typeimplementation) in the VANTIQ Modelo IDE. For an example of the definition, 
+please see the [*hikVisionImpl.json*](src/test/resources/hikVisionImpl.json) file located in the *src/test/resources* directory.
+
+Additionally, an example project named *HikVisionExample.zip* can be found in the *src/test/resources* directory.
+
+``` 
+
+{
+   "general":{
+   		"sdkLogPath": "c:/tmp/log",
+   		"DVRImageFolderPath": "c:/tmp/Thermo",
+   		"VantiqDocumentPath": "public/image",
+   		"VantiqResourcePath": "/resources/documents"},
+   "cameras": [
+      {
+         "CameraId": "C1",
+         "Enable": "true",
+         "DVRIP": "192.168.1.2",
+         "DVRPort": "8000",
+         "DVRUserName": "admin",
+         "DVRPassword": "pass1"
+      },
+      {
+         "CameraId": "c2",
+         "Enable": "false",
+         "DVRIP": "realhikvisionthermal.dyndns.org",
+         "DVRPort": "8000",
+         "DVRUserName": "admin",
+         "DVRPassword": "pass2"
+      },
+   ],
+   "options": {
+      "maxActiveTasks": 3,
+      "maxQueuedTasks": 20
+   }
+}
+```
+
+### Configuration
+
+*   **sdkLogPath**: Optional. the folder where the HikVision sdk log files are going to be created , default value `c:\tmp\log`
+*   **DVRImageFolderPath**: Optional . The prefix of the file pattern to look for , if not set an empty string will be used as default value `c:/tmp/Thermo`
```

Comment:
this says that if not set, it's empty but then gives a default path?  Is this a path or a file name prefix?
```suggestion
*   **DVRImageFolderPath**: Optional . The prefix of the file pattern to look for. if not set an empty string will be used as default value `c:/tmp/Thermo`
```